### ISSUE TITLE
Instant Stomach targeting, Godly Bot nades, Lock on through walls.

### ISSUE THREAD

**Issue by MKrameUltra** [Solarint/SAIN#132]
Game version - 3.8.3 
-Mods - 
Fika server
Server Value Modifier
Fontain FOV fix
zSolarint SAIN Server mod w/ Big Brain + Waypoints
Skwizzy no discard limit
Realistic Thermals 1.0.3
WTT Sianyde And G3iness
zrairaitheraichu ammostats 3.1.1
Amands Graphics

My SAIN settings are mainly default with the exception of the "Headshot Protection", "Bot Reaction and Accuracy changes toggle - Experimental", and "No Bush ESP" turned on (I forget if No Bush ESP is on by default".
![EscapeFromTarkov_qh5wmsMXmr](https://github.com/Solarint/SAIN/assets/171216310/1b57132d-ac2d-45b8-8384-e624eca9feac)

Server Value Modifier has "Player Scav cooldown" lowered, and Raid Time increased.

I've been having a problem with particular bot behaviors around Stomach targeting, lock-on through walls, and bot nades ( I just turned off bot nades for now)

Stomach targeting and instant blacking -
Bots are instant targeting/blacking my stomach regardless of armor/cover and ammo type instantly, and I am unsure if this is related to the "Headshot Protection" feature or not. 
Over the last 2 weeks of my playing on 3.8.3 the number one hit spot on my character is the stomach, and it becomes instantly black, and typically from unsuspecting ammo and armor conditions. For one example, on interchange I was using the large planter/dividers that line the hallways around Techlight as cover to shoot at an enemy at Burgerhouse / Sushi Huyushi and my stomach was blacked by a 9mm PSO despite the cover and the NFM Thor body armor.
![EscapeFromTarkov_LBLuajSGsT](https://github.com/Solarint/SAIN/assets/171216310/850790c4-f7e3-4ddb-8cea-e22270c4c449) 
Pic shows the cover I referenced. 

Bots locking on through walls - 
Most noticeable on Factory. Bots often are fighting each other through portions of the map they cannot see, but track each other through the walls. 
One example is when I flash banged a "PMC" that was hiding in a room, they did their usual 'random fire burst' but as I began moving towards the room to enter, they were tracking me through the wall with gun fire. 

Bot nades -
Basically God nades everytime, they mortar them over objects and direct impact perfectly despite no visibility of my character. They are able to sprint while throwing them making it difficult to predict, and I just had one throw a nade while not facing me on interchange which really made me determined to come here (was a great raid up until then lol).

Sorry for the mountain of text, I just wanted to be as detailed as possible.

**Comment by Kuromori117** [comment]
I've had many separate instances of bots having "esp" for lack of a better descriptor ever since the update last week. Just today on both customs raids I did, any bot near dorms will come to your exact floor/room no matter how quiet I am once I fire my weapon, even upon repositioning to a different room etc. 2nd one happened after I killed a scav on 2nd floor of 3-story, I then ran to a room at the end of the hallway to repack mags. not two minutes later two pmc's fight outside the door and then the winner proceeds to immediately come into the room I was in at the end of the hall. hadn't made a noise since walking there unless packing mags now makes noise? All of this on bog standard Default preset with aggression turned down by .1, and with Looting Bots and Realism the only other ai-changing mods.

  What I think is going on is bots are getting TOO much info from sound, regardless of player-made or not, to the point they pinpoint your exact xyz coordinates instead of 'vague 4-8m area'. It was DEFINITELY not like this before the new hearing implementation.
  
  Before the comments were turned off a few days ago on the Sain Mod Page there were plenty of people commenting on esp bots suddenly being a thing and Solarint dismissing as "just reposition", but being in a large building, firing your gun twice and reloading shouldnt notify every bot in the vicinity of your exact xyz coord, bot hearing is just way overtuned at the moment.
  
  Edit* I see the recent betas are messing more with hearing and cqb so the esp stuff might be addressed soon, thanks Solarint for your hard work!

**Comment by Somesphinx** [comment]
I want to add to that, that the stomach targeting and seemingly ignoring of armor covering that body part suddenly stopped for me. I wasn't paying too much attention to my level, but it must have been around level 15, when due to me leveling up the bot generation also advanced. Happened when playing solo with a lot of mods and again playing together via fika with only SAIN+BIGBRAIN+WAYPOINTS and Amands Graphics.

Edit* Thanks for your awesome work!

Edit two weeks later** So, hundreds of raids later now, I gotta say, it did not stop. It just happens less frequently and I got lucky for a while or whatever.

**Comment by Solarint** [comment]
A lot of bot vision got revamped in 3.0.0. Give it a shot.
